x/evm/keeper/send.go

```diff
@@ -0,0 +1,55 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/sei-protocol/sei-chain/x/evm/types"
+)
+
+func (k *Keeper) EVMToEVMSend(ctx sdk.Context, fromAddress common.Address, toAddress common.Address, amount uint64) error {
+	if err := k.DebitAddress(ctx, fromAddress, amount); err != nil {
+		return err
+	}
+	return k.CreditAddress(ctx, toAddress, amount)
+}
+
+func (k *Keeper) BankToEVMSend(ctx sdk.Context, bankAddress sdk.AccAddress, toAddress common.Address, amount uint64) error {
+	// first transfer tokens to evm module account
+	if err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, bankAddress, types.ModuleName,
+		sdk.NewCoins(sdk.NewCoin(k.GetBaseDenom(ctx), sdk.NewIntFromUint64(amount)))); err != nil {
+		return err
+	}
+	// then credit the recipient address
+	return k.CreditAddress(ctx, toAddress, amount)
+}
+
+func (k *Keeper) EVMToBankSend(ctx sdk.Context, evmAddress common.Address, toAddress sdk.AccAddress, amount uint64) error {
+	// first transfer tokens out of evm module account
+	if err := k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, toAddress,
+		sdk.NewCoins(sdk.NewCoin(k.GetBaseDenom(ctx), sdk.NewIntFromUint64(amount)))); err != nil {
+		return err
+	}
+	// then debit the sender address
+	return k.DebitAddress(ctx, evmAddress, amount)
+}
+
+func (k *Keeper) CreditAddress(ctx sdk.Context, address common.Address, amount uint64) error {
+	balance := k.GetBalance(ctx, address)
+	if balance+amount < balance {
```

Comment:
is this intended to check for number wrap around? we could probably just compare balance to UINT64.MAX or something?

Reply:
balance itself is by invariant always valid, but to enforce that invariant we need to check the amount we set to balance which is `balance+amount`. I've changed it to a more readable check for this purpose